### ISSUE TITLE
[mobile app] Vaccination status is not auto-set  to "vaccinated" for relevant vaccinations of primary and secondary cases

### ISSUE THREAD

**Issue by BasilKaburi** [hzi-braunschweig/SORMAS-Project#8114]
### Bug Description

Vaccination status is not auto-set to "vaccinated" for relevant vaccinations of primary cases and those resulting from event participants and contacts.
### Steps to Reproduce

Consider a vaccinated confirmed contact that is converted to a case:
1. Create a new contact and set the report date to the curent date
2. Set the contact to confirmed
3. Add a vaccination and set the vaccination date a week prior to the report date
4. Convert the contact into a case
5. Check vaccination entries and vaccination status2.

### Expected Behavior

The vaccination status should be set to vaccinated.
### Screenshots

![Screenshot_20220221-124121_APMIS](https://user-images.githubusercontent.com/71414361/155411882-3c52b6d3-951f-4d15-a87b-644052d5c491.jpg)### System Details

* Device: Android
* SORMAS version: 1.68.0
* Android version/Browser: Version 11 (knox 3.7, API level 33)
* Server URL: https://release-international.sormas.netzlink.com
* User Role: Surveillance officer & Contact officer

**Comment by dinua** [comment]
The status to "vaccinated" is set only after the contact or case are synchronized with the server . This operation is done in backend.

**Comment by adinaflorea9** [comment]
Verified ticket on APK download: https://test.sormas.netzlink.com/downloads/release/ ; Server URL: https://test.sormas.netzlink.com/sormas-rest/ using the SORMAS version 1.71.0-SNAPSHOT.

The status 'Vaccinated' is set for entities after the synchronization with the server.

**Comment by BasilKaburi** [comment]
@adinaflorea9, @MateStrysewske  : From my tests, this issue failed on production, I'm afraid.

Device: Android
SORMAS version: 1.71.0
Android version/Browser: Version 11 (knox 3.7, API level 33)
Server URL: https://release-international.sormas.netzlink.com.sormas-rest/
User Roles (2): Surveillance officer & Contact officer, and hospital informant.

Please, could you take a second look?

**Comment by MateStrysewske** [comment]
@BasilKaburi If this is not working on 1.71, please open a new bug report so that we can investigate it.

**Comment by adinaflorea9** [comment]
@BasilKaburi I am currently re-testing this on my local machine. Will let you know if it is reproducible there or not.

**Comment by BasilKaburi** [comment]
@MateStrysewske In my opinion (which is often narrow in these matters), this will amount to duplicating the issue. If the development of an issue fails on first attempt, I  guess we should not consider it a new issue. Instead, I suggest we reopen the issue and continue investigations to fix it. If I'm wrong, I will just create a duplicate issue.

@adinaflorea9 It's fine to re-test on your local machine. However, the ultimate pass should be a pass on the production server: 
 https://release-international.sormas.netzlink.com.sormas-rest/
Could you also try to reproduce it on this release instance?
Fingers crossed :)

**Comment by adinaflorea9** [comment]
@BasilKaburi re-tested the behavior reported in this ticket both on my local machine and on https:release-international.sormas.netzlink.com.

I tested both via full immunization module and via the reduced immunization module. The reported behavior is not reproducible on either one of these 2 test instances anymore, if all of the reported steps are done and the immunization status is checked AFTER the synchronization with the server is successfully completed (as @dinua highlighted on the 29th of March).

**Comment by BasilKaburi** [comment]
@adinaflorea9 Thanks for updates and clarifications. I will test with 1.72.1.

**Comment by adinaflorea9** [comment]
@BasilKaburi Glad to have helped. Just as an extra note - the SORMAS version on my local today was 1.73.0; and on release-international it was 1.71.0.

**Comment by BasilKaburi** [comment]
@adinaflorea9 Right! Thanks!

**Comment by MateStrysewske** [comment]
@BasilKaburi The reason for my request is that this issue here has already been released, and we're not re-opening issues that have already been released because it would cause more confusion than a duplicate issue (since one issue can only be related to one milestone). Adina has given a very good indication here though in that synchronization should be done before checking whether the issue in fact still persists.

**Comment by BasilKaburi** [comment]
@MateStrysewske Thanks very much for the explanation. It makes absolute sense. I will test on 1.72.1 and share feedback.

**Comment by BasilKaburi** [comment]
@MateStrysewske : For 1.72.1 and 1.73.1 I did not get expected result as you explained. However, I have no way of ascertaining that synchronisation and other pending performance issues are not contributing to the apparent failure.
@adinaflorea9 Please, did you use the exact steps outlined in this issue? Otherwise, I would like to retest with your test steps. It could just a be question of a missing link in the steps.

**Comment by adinaflorea9** [comment]
Following @BasilKaburi 's latest comment, I have retested the behavior both on my local machine and on a mobile device (Android 9 - Huawei P20 Lite). As I no longer have access to the SORMAS version mentioned by Basil in the latest comment, I have used the following versions of SORMAS to retest this: 
- on the local machine - SORMAS version 1.75.0-SNAPSHOT(0a3c144);
- on release-international.sormas.netzlink.com - SORMAS 1.73.2;

Users used for testing were Surveillance Officers.

In my comment https://github.com/hzi-braunschweig/SORMAS-Project/issues/8114#issuecomment-1137295082 from the 25th of May I noted that at the time I had re-tested for a second time the behavior on SORMAS versions 1.71.0  and 1.73.0, where the behavior was no longer reproducible for me. 

Please note and take into consideration when testing Adrian's comment https://github.com/hzi-braunschweig/SORMAS-Project/issues/8114#issuecomment-1081590083 from the 29th of March - without the synchronization being triggered the vaccination status of entities is not properly updated.

Also, please note that the vaccinations have to be considered relevant in order for the vaccination status to be updated to 'vaccinated' (i.e. that the time frame between the report dates of the entities and the 'report dates of the immunization' + 'report dates of the vaccinations' + 'vaccination date' is valid).

**Scenarios that were covered during re-testing:**

**A: release-international.sormas.netzlink.com - SORMAS version is 1.73.2 - full immunization module is enabled**

I: Case in the mobile app:
1. Open the 'New case' form;
2. Fill in the mandatory information (disease: Covid-19; case report date: 1st of July);
3. Save the case;
4. Navigate to the 'Immunizations' tab of the case;
5. Open the 'New immunization' form, input the mandatory information (date of report: 24th of June; disease: Covid-19; means of immunization: vaccination);
6. Save the immunization;
7. Navigate to the 'Vaccinations' tab of the immunization;
8. Open the 'New vaccination' form and fill in the following information: date of report: 24th of June, vaccination date: 24th of June; vaccine name;
9. Save the vaccination and trigger a synchronization with the server;
10. Observe the vaccination status of the case (=> 'vaccinated' both in the mobile-app and in the web-app).

II: Contact + Contact's resulting case in the mobile app:
1. Open the 'New Contact' form;
2. Fill in the mandatory information (disease: Covid-19; report date: 1st of July);
3. Save the contact;
4. Navigate to the 'Contact Immunizations' tab;
5. Open the 'New Immunization' form  and fill in the mandatory information (date of report: 24th of June; disease: Covid-19; means of immunization: vaccination);
6. Save the immunization;
7. Navigate to the 'Vaccinations' tab of the immunization;
8. Open the 'New vaccination' form and fill in the following information: date of report: 24th if June; vaccination date: 24th of June; vaccine name;
9. Savte the vaccination and trigger a synchronization with the server;
10. From the contact's page, set the 'Contact classification' to 'Confirmed' and save;
11. From the contact's page, click the 'Create a case for this contact person' button;
12. In the 'New case' form fill in the mandatory information (date of report: 1st of July);
13. Save the case;
14. Trigger a synchronization with the server;
15. Observe the vaccination status of the case and the contact (=> 'vaccinated' both in the mobile-app and in the web-app).


III: Event participant + Event participant's resulting case in the mobile app:
1. Open the 'New event' form and fill in the mandatory information (disease: Covid-19, date of event and date of report: 1st of July);
2. Save the event;
3. Open the 'New person involved' form (aka the new event participant form) and fill in the mandatory information;
4. Save the event participant;
5. Navigate to the 'Event Participant immunizations' tab;
6. Open the 'New immunization form' and fill in the mandatory information (date of report: 24th of June; means of immunization: vaccination);
7. Save the immunization;
8. Navigate to the 'Vaccinations' tab of the immunization;
9. Open the 'New vaccination' form and fill in the following information: date of report: 24th of June, vaccination date: 24th of June, vaccine name;
10. Save the vaccination and trigger a synchronization with the server;
11. Navigate to the event participant's page and click the 'Create case for this event person' button;
12. In the 'New case' form fill in the mandatory information (date of report: 1st of July);
13. Save the case and trigger  a synchronization with the server;
14. Observe the vaccination status of the case and the event participant (=> vaccinated both in the mobile-app and in the web-app)


**B: local machine - SORMAS version is 1.75.0 - where the reduced immunization module is enabled**

I: Case in the mobile app:
1. Open the 'New case' form;
2. Fill in the mandatory information (disease: Covid-19; case report date: 1st of July);
3. Save the case;
4. Navigate to the 'Case Vaccinations' tab;
5. Open the 'New vaccination' form and fill in the following information: date of report: 24th of June, vaccination date: 24th of June, vaccine name;
6. Save the vaccination and trigger a synchronization with the server;
7. Observe the vaccination status of the case (=> 'vaccinated' both in the mobile-app and in the web-app).


II: Contact + Contact's resulting case in the mobile app:
1. Open the 'New Contact' form;
2. Fill in the mandatory information (disease: Covid-19; report date: 1st of July);
3. Save the contact;
4. Navigate to the 'Contact Vaccinations' tab;
5. Open the 'New Vaccination' form  and fill in the following information: date of report: 24th of June, vaccination date: 24th of June, vaccine name;
6. Save the vaccination and trigger a synchronization with the server;
7. From the contact's page, set the 'Contact classification' to 'Confirmed' and save;
8. From the contact's page, click the 'Create a case for this contact person' button;
9. In the 'New case' form fill in the mandatory information (date of report: 1st of July);
10. Save the case and trigger a synchronization with the server;
11. Observe the vaccination status of the case and the contact (=> 'vaccinated' both in the mobile-app and in the web-app).


III: Event participant + Event participant's resulting case in the mobile app:
1. Open the 'New event' form and fill in the mandatory information (disease: Covid-19, date of event and date of report: 1st of July);
2. Save the event;
3. Open the 'New person involved' form (aka the new event participant form) and fill in the mandatory information;
4. Save the event participant;
5. Navigate to the 'Event Participant vaccinations' tab;
6. Open the 'New vaccination' form and fill in the following information: date of report: 24th of June, vaccination date: 24th of June, vaccine name;
7. Save the vaccination and trigger a synchronization with the server;
8. Navigate to the 'Vaccinations' tab of the immunization;
9. Navigate to the event participant's page and click the 'Create case for this event person' button;
10. In the 'New case' form fill in the mandatory information (date of report: 1st of July);
11. Save the case and trigger  a synchronization with the server;
12. Observe the vaccination status of the case and the event participant (=> vaccinated both in the mobile-app and in the web-app)

@BasilKaburi if the behavior is still reproducible for you, please open a separate bug report with the exact detailed steps that you are following.

**Comment by BasilKaburi** [comment]
@adinaflorea9 @MateStrysewske 
I'm excited to report that the test passed.
I just tested with 1.74.1.
Thanks very much for the detailed steps and hints; and your patience also!

**Comment by adinaflorea9** [comment]
@BasilKaburi I am happy to have been able to help and that the tests now pass!
